Lib/email/utils.py

```diff
@@ -103,12 +105,68 @@ def formataddr(pair, charset='utf-8'):
     return address
 
 
+def getaddresses(fieldvalues, *, strict=True):
+    """Return a list of (REALNAME, EMAIL) or ('','') for each fieldvalue.
+
+    When parsing fails for a fieldvalue, a 2-tuple of ('', '') is returned in
+    its place.
+
+    If strict is True, use a strict parser which rejects malformed inputs.
+    In this case, if the resulting list of parsed addresses is greater than
+    the number of fieldvalues in the input list, a parsing error has occurred
+    and consequently a list containing a single empty 2-tuple [('', '')] is
+    returned in its place. This is done to avoid invalid output.
 
-def getaddresses(fieldvalues):
-    """Return a list of (REALNAME, EMAIL) for each fieldvalue."""
-    all = COMMASPACE.join(str(v) for v in fieldvalues)
+    Malformed input: getaddresses(['alice@example.com <bob@example.com>'])
+    Invalid output: [('', 'alice@example.com'), ('', 'bob@example.com')]
+    Safe output: [('', '')]
+
+    """
+    if not strict:
+        all = COMMASPACE.join(str(v) for v in fieldvalues)
+        a = _AddressList(all)
+        return a.addresslist
+
+    fieldvalues = [str(v) for v in fieldvalues]
+    fieldvalues = _pre_parse_validation(fieldvalues)
+    all = COMMASPACE.join(v for v in fieldvalues)
     a = _AddressList(all)
-    return a.addresslist
+    result = _post_parse_validation(a.addresslist)
+
+    # When a comma is used in the Real Name part it is not a deliminator
+    # So strip those out before counting the commas
+    n = 0
+    for v in fieldvalues:
+        v = realname_comma_re.sub('', v)
```

Comment:
What if the input is `'"Jane Doe" <jane@example.net>, "John Doe" <john@example.net>'`?

Reply:
The `realname_comma_re` regex is a weak parser for an email which accepts to escape quotes. I wrote `_iter_escaped_chars()` and `_strip_quoted_realnames()` instead and added tests on them.